lib/vsc/utils/nagios.py

```diff
@@ -277,7 +277,11 @@ def report_and_exit(self):
             unknown_exit("%s nagios gzipped JSON file unavailable (%s)" % (self.header, self.filename))
 
         (timestamp, ((nagios_exit_code, nagios_exit_string), nagios_message)) = nagios_cache.load('nagios')
+        self.print_report_and_exit(timestamp, nagios_exit_code, nagios_exit_string, nagios_message)
 
+
+    def print_report_and_exit(self, timestamp, nagios_exit_code, nagios_exit_string, nagios_message):
+        """Print the nagios report (if the data is not too old) and exit"""
```

Comment:
why do you not need/use the if/else logic? i would have expected that you only want to override the if block?

Reply:
for zabbix the if/else logic is skipped: we always report, even if the data is too old, and let zabbix handle that.